packages/dd-trace/src/config.js

```diff
@@ -163,6 +164,36 @@ class Config {
       process.env.DD_TRACE_EXPERIMENTAL_TRACEPARENT_ENABLED,
       false
     )
+    const DD_TRACE_PROPAGATION_STYLE_INJECT = coalesce(
+      options.tracePropagationStyle && options.tracePropagationStyle.inject,
+      process.env.DD_TRACE_PROPAGATION_STYLE_INJECT,
+      // TODO: change this to traceparent,datadog
+      // This can't be done as a semver-minor until tracestate support also exists
+      'datadog'
+    ).split(',').reduce((m, v) => {
+      m[v.toLowerCase()] = true
+      return m
+    }, {
+      b3: isTrue(DD_TRACE_B3_ENABLED),
+      'b3 single header': isTrue(DD_TRACE_B3_ENABLED),
+      tracecontext: isTrue(DD_TRACE_TRACEPARENT_ENABLED),
+      datadog: false
```

Comment:
Why does everything have a dynamic default except this? Should everything just be false by default as soon as this option is used?

Reply:
Everything should be false. The dynamic defaults are just to allow the experimental flags to continue to work as they used to. Alternatively, we can just hard-drop those configs if we prefer. Just seemed like not much effort to keep support for the old configs.

Reply:
IMO the old config should only be used when the new config is not set. When the new config is set it should completely take over since it's not quite backward compatible in the sense that it has to manage priority too.